deltachat-ios/Helper/ImageFormat.swift

```diff
@@ -84,25 +84,55 @@ extension ImageFormat {
         return saveImage(data: data, name: name, suffix: suffix)
     }
 
-    public static func saveImage(data: Data, name: String? = nil, suffix: String) -> String? {
+    public static func saveImage(data: Data, name: String? = nil, suffix: String, directory: FileManager.SearchPathDirectory = .applicationSupportDirectory) -> String? {
         var path: URL?
-        guard let directory = try? FileManager.default.url(for: .documentDirectory, in: .userDomainMask,
-                                                              appropriateFor: nil, create: false) as NSURL
-        else { return nil }
+
+        // ensure directory exists (application support dir doesn't exist per default)
+        let fileManager = FileManager.default
+        let urls = fileManager.urls(for: directory, in: .userDomainMask) as [URL]
+        guard let identifier = Bundle.main.bundleIdentifier else {
+            print("err: Could not find bundle identifier")
+            return nil
+        }
+        guard let directoryURL = urls.last else {
+            print("err: Could not find directory url for \(String(describing: directory)) in .userDomainMask")
+            return nil
+        }
+        var subdirectoryURL = directoryURL.appendingPathComponent(identifier)
```

Comment:
why are we creating an extra subdirectory here? isn't applicationSupportDirectory already unique to our app?

in general, maybe add  a some link to some documentation or used source for this whole block.

Reply:
I sould stick to the recommendations in this case: https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/AccessingFilesandDirectories/AccessingFilesandDirectories.html
the identifier is generally used to avoid file collisions. I'll add a comment